package/MDAnalysis/lib/formats/xdrlib.pyx

```diff
@@ -0,0 +1,653 @@
+    int xdrfile_close (XDRFILE * xfp)
+    int xdr_seek(XDRFILE *xfp, int64_t pos, int whence)
+    ctypedef float matrix[3][3]
+    ctypedef float rvec[3]
+
+
+cdef extern from 'include/xdrfile_xtc.h':
+    int read_xtc_natoms(char * fname, int * natoms)
+    int read_xtc(XDRFILE * xfp, int natoms, int * step, float * time, matrix box,
+                 rvec * x, float * prec)
+    int write_xtc(XDRFILE * xfp, int natoms, int step, float time, matrix box,
+                  rvec * x, float prec)
+
+    int read_xtc_n_frames(char *fn, int *n_frames, int *est_nframes, int64_t **offsets);
+
+
+cdef extern from 'include/xdrfile_trr.h':
+    int read_trr_natoms(char *fname, int *natoms)
+    int read_trr(XDRFILE *xfp, int natoms, int *step, float *time, float *_lambda,
+                 matrix box, rvec *x, rvec *v, rvec *f, int *has_prop)
+    int write_trr(XDRFILE *xfp, int natoms, int step, float time, float _lambda,
+                  matrix box, rvec *x, rvec *v, rvec *f)
+    int read_trr_n_frames(char *fn, int *n_frames, int *est_nframes, int64_t **offsets);
+
+
+
+cdef extern from 'include/xdrfile_trr.h':
+    int read_trr_natoms(char *fname, int *natoms)
+    int read_trr(XDRFILE *xfp, int natoms, int *step, float *time, float *_lambda,
+                 matrix box, rvec *x, rvec *v, rvec *f, int *has_prop)
+    int write_trr(XDRFILE *xfp, int natoms, int step, float time, float _lambda,
+                  matrix box, rvec *x, rvec *v, rvec *f)
+
+cdef enum:
+    EOK = 0
+    EHEADER = 1
+    ESTRING = 2
+    EDOUBLE = 3
+    EINTEGER = 4
+    EFLOAT = 5
+    EUNSIGNEDINTEGER = 6
+    ECOMPRESSION = 7
+    ECLOSE = 8
+    EMAGIC = 9
+    EMEMORY = 10
+    EENDOFFILE = 11
+    ENOTFOUND = 11
+
+error_message = ['OK', 'header', 'string', 'double', 'integer',
+                 'float', 'unsigned integer', 'compression',
+                 'close', 'magic', 'memory', 'endoffile',
+                 'notfound']
+
+import cython
+import numpy as np
+from os.path import exists
+from collections import namedtuple
+
+np.import_array()
+
+ctypedef np.float32_t DTYPE_T
+DTYPE = np.float32
+cdef int DIMS = 3
+cdef int HASX = 1
+cdef int HASV = 2
+cdef int HASF = 4
+
+
+cdef class _XDRFile:
+    """Base python wrapper for gromacs-xdr formats
+
+    This class can be similar to the normal file objects in python. The read()
+    function will return a frame and all information in it instead of a single
+    line. Additionally the context-manager protocoll is supported as well.
+
+    Parameters
+    ----------
+    fname: str
+        The filename to open.
+    mode: ('r', 'w')
+        The mode in which to open the file, either 'r' read or 'w' write
+
+    Raises
+    ------
+    ValueError
+        Wrong mode given
+    IOError
+        Couldn't read the file
+
+    Note
+    ----
+    This class can't be initialized use one of the subclasses XTCFile, TRRFile
+    """
+    cdef readonly int n_atoms
+    cdef int is_open
```

Comment:
No these are not shared across all instances. They are used here to declare private struct members with a type. Unlike normal python-classes cython structs need to have all typed members declared at the beginning.